src/PushNotifications.php

```diff
@@ -1,12 +1,31 @@
 <?php
 namespace Pusher\PushNotifications;
+
+use GuzzleHttp;
+
+/**
+ * Pusher Push Notifications client class
+ * Used to publish notifications to the Pusher Push Notifications API
+ * http://www.pusher.com/push-notifications
+ */
 class PushNotifications {
   const SDK_VERSION = "0.10.5";
+  const MAX_INTERESTS = 100;
+  const MAX_INTEREST_LENGTH = 164;
+  const INTEREST_REGEX = "/^(_|-|=|@|,|\\.|:|[A-Z]|[a-z]|[0-9])*$/";
+
+  private $options = array();
+  private $client = null;
 
-  public function __construct($options) {
+  public function __construct($options, $client = null) {
```

Comment:
Hmm, couldn't the client be part of the `options`?

TBH, I'd like more if the `instanceId` and `secretKey` were dedicated arguments and then just a misc `options` for the rest.

Reply:
I'd like to do this in a separate PR as it would be a breaking change.